Improve a TAIT error and add an error code plus documentation (#106940)
rust-lang/rust: compiler/rustc_error_codes/src/error_codes/E0792.md

> Type alias impl trait can only have their hidden type set

Reviewer: ```suggestion
A type alias impl trait can only have its hidden type defined
```

Reviewer: or maybe "assigned"?

Author: I'll go with `assigned`. Opinions at https://github.com/rust-lang/reference/pull/1317/files#diff-5aaceb2a1fd048b1730a7dcaff82d85a05fd5c64faa4ad9df1773adfcd6cf2c1R75 are very welcome ^^
